envoy/network/listener_filter_buffer.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * Interface for ListenerFilterBuffer
+ */
+class ListenerFilterBuffer {
+public:
+  virtual ~ListenerFilterBuffer() = default;
+
+  /**
+   * Return a single const raw slice to the buffer of the data.
+   * @return a Buffer::ConstRawSlice pointed to raw buffer.
+   */
+  virtual const Buffer::ConstRawSlice rawSlice() const PURE;
+
+  /**
+   * Drain the data from the beginning of the buffer.
+   * @param length the length of data to drain.
+   * @return return True if the drain success, otherwise return False.
+   */
+  virtual bool drain(uint64_t length) PURE;
```

Comment:
Is the return value just for assertion checking? Presumably this can't fail unless someone tries to drain more data then is in the buffer? I'm wondering if this should return void, and either assert inside the function, or just drain the entire buffer <= length?

Reply:
This return value can be removed.

We have both assertion on the `recv` return value
https://github.com/envoyproxy/envoy/pull/17395/files#diff-1bc1bf13bcf1d909c2713dd218446fea73ba40e513720685ca7cfc3117ace8bbR43-R45

and on the drain size
https://github.com/envoyproxy/envoy/pull/17395/files#diff-1bc1bf13bcf1d909c2713dd218446fea73ba40e513720685ca7cfc3117ace8bbR36

I remember we have similar discussion on whether the drain can fail. Probably I added the assertion as the comment but forget to remove this return value.